src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -944,8 +948,16 @@ public Void caseObjectType(ObjectType type) {
           } else if (type.isDict()) {
             emit("{[key: string]: any}");
           } else if (type.getReferenceName() != null) {
+            String name;
+            // If a class A is declared in goog.module and exported as default, closure
+            // assigns it a synthetic type $jscomp.scope.A.
+            if (type.getReferenceName().contains("$jscomp.scope")) {
```

Comment:
The error actually looks like this:
`Property foo never defined on variable$$module$path$to$usage$file of type A$$module$path$to$definition$of$A`, so the path before `.` gets stripped, but the module path gets appended separately to the back.